### PR TITLE
New test to validate bug fix

### LINKED ISSUE #17: Number of examples changes in last batch in ClassificationTrainer when calling main.py multiple times.
One possible reason for this is that the validation split in DatasetFactory is not seeded. Although the split will always give the same number of total examples, it does not consistently give the same number of examples in each class. We are forcing an imbalance_ratio based on the number of examples in the first class. If it changes, the total number of examples will change. 

- [ ] Verify that the bug is indeed the seed in DatasetFactory. 
- [ ] Add seed directly in main that seeds all of torch.

### PR DESCRIPTION
**Pull request by Akssi**
closes #17

### CODE REVIEW

File: tests/test_dataset_factory.py [hunk 1]
```diff
@@ -24,6 +25,20 @@ def dataset_factory(self):
                                                                         batch_size=self.batch_size)
         yield train_loader, valid_loader, test_loader
 
+    def test_dataset_last_batch_has_proper_number_of_examples(self, dataset_factory):
+        _, valid_loader, _ = dataset_factory
+        valid_data, valid_labels = next(iter(valid_loader))
+        _, first_batch_counts = np.unique(valid_labels, return_counts=True)
+        while True:
+            try:
+                valid_data, valid_labels = next(dataloader_iterator)
```
Comment (Olivier-tl) [inline tests/test_dataset_factory.py:34]: Le linting a pas passé à cause de cette ligne la! (dataloader_iterator n'existe pas)

Très cool de faire un test pour ça btw! Mais je ne vois pas comme ça va passer par contre. C'est normale qu'il y aille moins d'example dans la dernière batch si le nombre total d'examples n'est pas un multiple de la batch size. Aussi, la tu check que la premiere & la derniere batch a exactement le même nombre de labels. J'pense pas que ce soit requis qu'il y aille exactement le même nombre de labels dans chaque batch. 

Dans le fond pour valider #17, on peut juste créer deux datasets et valider qu'ils ont le même nombre d'examples. Le seed est set dans main.py, alors il va falloir setter le seed dans le test, sinon je ne pense pas que ça va passer.

Comment (Akssi) [reply tests/test_dataset_factory.py:34]: image
See the comments in code. Basically, the len of the datasets are always the same and the last batch of valid is always of a different shape no matter if the seed is set or not.
